### ISSUE TITLE
Pointless `root' permission request & dubious package manipulation

### ISSUE THREAD

**Issue by lacombar** [JBakamovic/yavide#23]
I just attempted to install `yavide`, and the requirement for `root` privilege is unsettling. First of all, I don't trust scripts to run as `root`, cf the recent failure with Steam bug which cause havoc[0].

Reasonable practice should be:
 - `yavide` should not require an up-to-date system
 - `yavide` should not update the user system
 - `yavide`  should not require `root` permission

If dependencies are needed, explicitly tell the user to install the required packages.

If the destination directory is not user-writable, the user should choose to run the script using his preferred method under the relevant user.

[0]: http://www.pcworld.com/article/2871653/scary-steam-for-linux-bug-erases-all-the-personal-files-on-your-pc.html

**Comment by JBakamovic** [comment]
Right, most of the time I am focused on adding new features but this will be fixed as well. At the time when I was writing this installation script, idea was to ease the whole installation process as much as possible and not to bug users to install everything by hand. However, I do agree with aforementioned things you've pointed out.

**Comment by CDSRV** [comment]
the request for the user's password is a little cryptic as its not actually stating that the user running the script needs to be able to run 'sudo' and things don't seem to install correctly if the script is run as root.  also if its installing other packages it should also check for important dependencies such as 'gvim'

**Comment by mateuszlewko** [comment]
I 'd like to install this but I don't have root permissions on my workstation, what can I do?
I'm running CentOS 6.6

Thanks!

**Comment by JBakamovic** [comment]
So you are not able to run any `yum install` commands?

If not then I am afraid you will not get it run very easily. `Yavide` depends on [variety of libraries](https://github.com/JBakamovic/yavide/blob/master/install.sh#L50). However, most of these are pulled in because of ctags/cscope/python-watchdog and my plan is to completely remove need for them. This will happen as soon as I finish with the implementation of clang-based indexer.

**Comment by evg-zhabotinsky** [comment]
This issue really needs to be addressed. Using root unnecessarily is bad. `make install` is usually separate for a reason, and even then it's highly discouraged to run it as root on real system.

Right now you are using root to clone repos and pull stuff with `git`, and to download, unpack and build `cppcheck`. One should never do such things as root.

You also run `make install` for `cppcheck` as root, which is generally a really bad idea too. If you really need a fresh version, you could install a private copy under `yavide` directory as non-root user.

As a side note, using `pip install` as root totally wrecked my distro-provided (I use Linux Mint) Python installation once, to the point of Cinnamon desktop periodically freaking out and some apps like Deluge or even IPython qtconsole not starting at all. I had to manually scrub every mention of Python from my filesystem and then reinstall all Python-related packages. It's better to use packages from distro repos instead, and only after that resort to `pip install`. Or even better, run `pip` as normal user to install packages somewhere under `$HOME`.

And finally, it shouldn't be that hard to properly package this app. I don't have much experience with building `.deb` packages, but `.rpm` one is about as easy as reformatting your `install.sh` a bit to match `.spec` file syntax.

**Comment by JBakamovic** [comment]
> Right now you are using root to clone repos and pull stuff with git, and to download, unpack and build cppcheck. One should never do such things as root.

You somehow missed the necessity of [system packages](https://github.com/JBakamovic/yavide/blob/master/install.sh#L50) installation? That is the reason for root access and it is not unnecessary unless you point me to a better way?

**Comment by lacombar** [comment]
There is 21 command using `sudo`, and you're justifying its use by a use-case representing 5% of that list ?

**Comment by lacombar** [comment]
... and btw, I can understand devel libs, but `gvim` ? why ?

**Comment by JBakamovic** [comment]
If you feel so please contribute with a better solution. You seem to be very knowledgeable on the topic ... perhaps better than me.

But I am glad that we are finally on the same page and that we agree it's not so "pointless" anymore.

**Comment by evg-zhabotinsky** [comment]
I think I intend to fix packaging issue. Got a question though:
Why exactly are you rebuilding cppcheck? It should be available in repos.

**Comment by JBakamovic** [comment]
At the time when I was initially writing this script, `cppcheck` was not available in the repos so yeah it is a little bit outdated (and also wrong because it can overwrite your system `cppcheck` installation).
